### ISSUE TITLE
Using explicit constructor for DynamicParameters type causes DAP214 error

### ISSUE THREAD

**Issue by oriches** [DapperLib/DapperAOT#48]
Using something similar to the following causes a couple of DAP214 variable not declared errors, when infact the code works as expected (results returned):

```
var parameters = new DynamicParameters();
parameters.Add("@StartDate", "2023-06-01");
parameters.Add("@EndDate", "2023-06-30");
```

When I switch to an anonymous type for the parameters everything is fine - no errors.
```
var parameters = new 
{
   parameters.Add("@StartDate", "2023-06-01").
   parameters.Add("@EndDate", "2023-06-30")
}
```

**Comment by mgravell** [comment]
OK; I'm going to suppress the parameter sniffing for `DynamicParameters`, and *add* a suggestion to use an anonymous type instead (they're much more efficient)

**Comment by mgravell** [comment]
I assume you mean:

``` c#
var parameters = new 
{
   StartDate = "2023-06-01",
   EndDate = "2023-06-30",
};
```

?
```

**Comment by oriches** [comment]
I did- thanks

**Comment by mgravell** [comment]
https://github.com/DapperLib/DapperAOT/commit/df6d435adafaa8392a5c28128e73bb66efccaccd and https://github.com/DapperLib/DapperAOT/issues/51

**Comment by mgravell** [comment]
fixed in 0.5.0-beta.104
